link/syscalls.go

```diff
@@ -61,7 +61,10 @@ var haveProgAttachReplace = internal.NewFeatureTest("BPF_PROG_ATTACH atomic repl
 		},
 	})
 	if err != nil {
-		return internal.ErrNotSupported
+		if errors.Is(err, unix.EINVAL) {
```

Comment:
Nit: `errors.Is` does checking for `nil`, so the following is equivalent and has less drift to the right:

```
if errors.Is(...) {
}
if err != nil {
}
```

In https://github.com/cilium/ebpf/issues/891#issuecomment-1535891636 I wrote:

> Note that errors.Is(err, EINVAL) is basically the same as checking err != nil, since EINVAL is such a common error that it's hard to infer anything.

Do you think that explicitly checking EINVAL here is meaningful?

Reply:
Not really, I believe it does not necessarily carry any extra value but I believe it's the closest error to return if that syscall was not supported by the kernel? See: [here](https://elixir.bootlin.com/linux/latest/source/kernel/bpf/syscall.c#L3521)

Reply:
> Not really, I believe it does not necessarily carry any extra value

Yeah, makes sense. Maybe we stick to the current behaviour then? It means less churn.

> See: [here](https://elixir.bootlin.com/linux/latest/source/kernel/bpf/syscall.c#L3521)

Nit: the EINVAL is returned here: https://elixir.bootlin.com/linux/v6.4.1/source/kernel/bpf/syscall.c#L5167

Reply:
Ah. Got it